File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int i=0,l=0,r=0,ans=0,len=0,size=0;
        while(r<nums.size()){
            if(nums[r] == 0){
                size++;
            }
            while(size>k){
             
              if(nums[l] == 0)
              size--;
              l++;
            }
            if(size<=k){
            len = r-l+1;
            ans = max(ans,len);
            }
            
          
            r++;
        }
        return ans;
    }
};

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int ans = 0,f = 0,end = 0,zero = 0;
        // while(end < nums.length){
        //   if(k == 0 && nums[end] == 0){
        //    while(k <= 0){
        //     if(nums[f] == 0){
        //         k++;
        //     }
        //     f++;
        //    }
        //   }
        //     if(nums[end] == 0){
        //     k--; 
        //    }
            
        //     ans = Math.max(ans,end - f +1);

        //     end++;
           
         
        // }

        while(end < nums.length){
             if(nums[end] == 0) zero++;
              if(zero > k){
                if(nums[f] == 0){
                    zero--;
                }
                f++;
              }
              if(zero<=k){
                ans = Math.max(ans,end-f+1);
              }
              end++;
             
        }
        return ans;
    }
}